internal/clientconn/conninfo/conn_info.go

```diff
@@ -57,6 +58,22 @@ func (connInfo *ConnInfo) SetAuth(username, password string) {
 	connInfo.password = password
 }
 
+// ClientMetadataPresence returns stored client metadata presence.
+func (connInfo *ConnInfo) ClientMetadataPresence() bool {
+	connInfo.rw.RLock()
+	defer connInfo.rw.Unlock()
```

Comment:
```suggestion
	defer connInfo.rw.RUnlock()
```

That's why it panics